test/workers/delete_plain_object_worker_test.rb

```diff
@@ -103,6 +103,7 @@ def load_target
     end
 
     def test_race_condition
+      skip('skip the test until we upgrade to Rails >= 7.1.0 that fixes https://github.com/rails/rails/pull/46553')
```

Comment:
I haven't reviewed this test, but couldn't we use `Thread` instead of `Fiber` in the test? so we don't need to skip it.

Reply:
I don't know... I thought it was too complicated :grimacing: 
But I can take a look.

There are some places where Threads will not be equivalent to Fibers, because we store some info in the current thread... But I am not actually sure if it's relevant to this test.

Reply:
I've taken a look at it, and I think it's not easy...

This tests hacks into the `delete` method to return the execution from the fiber: https://github.com/3scale/porta/blob/5d652f30763dafe5594ded648ea24b62bc65f866/test/workers/delete_plain_object_worker_test.rb#L98

So, the execution path is (as I understand it):
1. fiber 1 is started
2. in the middle of the execution (`delete` method) control is returned from fiber 1
3. fiber 2 is started
4. fiber 1 is resumed and completes its execution

I don't think we have this amount of control with Threads.

Also, I checked if we could monkey-patch the fix, but it's also tricky, because the fix is applied on top of Rails 7, and we're still on 6,  and also I believe this would affect all DB connections, not only tests.

I don't know... I think it's not worth the effort - especially if we don't touch the implementation of this worker :grimacing: 

But if somebody else wants to look into it - feel free :)

Reply:
Right, thanks for checking

Reply:
but please make the skip optional
```suggestion
      skip('skip the test until we upgrade to Rails >= 7.1.0 that fixes https://github.com/rails/rails/pull/46553') if Rails.gem_version < "7.1"
```

Reply:
while on it, we can create a `Queue` and make one thread pull from it which will block it. Then another thread can push something to the `Queue` and will get first thread continue. But fine with me with a conditional skip.

Reply:
> while on it, we can create a `Queue` and make one thread pull from it which will block it. Then another thread can push something to the `Queue` and will get first thread continue. But fine with me with a conditional skip.

Maybe let's do it in another PR?